### DISCUSSION TITLE
Are white balance and brightness adjustments supported?

### DISCUSSION THREAD

**Discussion by DevPlayNew** [HaishinKit/HaishinKit.swift#1553] [Q&A]
Always appreciate your support and would like to use the filter function in the code to support white balance and additional adjustments? Thank you.

**Comment by shogo4405** [comment]
No, there isn't. Since it's an iOS API, I assume it will be used from there.
- https://developer.apple.com/documentation/avfoundation/avcapturedevice/white_balance

You can get `AVCaptureDevice`:
```
guard let device = stream.videoCapture(for: 0)?.device else {
  return
}
do {
  try device.lockForConfiguration()
  device.unlockForConfiguration()
} catch {
}
```

They might be expecting a beautiful whitening filter, but there isn't one.
